addons/web/static/tests/views/fields/timezone_mismatch_field_tests.js

```diff
@@ -0,0 +1,175 @@
+/** @odoo-module **/
+
+import { click, editSelect, getFixture } from '@web/../tests/helpers/utils';
+import { makeView, setupViewRegistries } from '@web/../tests/views/helpers';
+
+let target;
+
+QUnit.module('Fields', (hooks) => {
+    hooks.beforeEach(() => {
+        target = getFixture();
+        setupViewRegistries();
+    });
+
+    QUnit.module('TimezoneMismatchField');
+
+    QUnit.test('widget  in a list view', async function (assert) {
+        assert.expect(5);
+
+        const serverData = {
+            models: {
+                partner: {
+                    fields: {
+                        color: {
+                            type: 'selection',
+                            selection: [
+                                ['red', 'Red'],
+                                ['black', 'Black'],
+                            ],
+                            default: 'red',
+                        },
+                        tz_offset: {
+                            string: 'tz_offset',
+                            type: 'char',
+                        },
+                    },
+                    records: [
+                        { id: 1, color: 'red', tz_offset: 0 },
+                        { id: 2, color: 'red', tz_offset: 0 },
+                        { id: 3, color: 'red', tz_offset: 0 },
+                    ],
+                    onchanges: {
+                        color: function (r) {
+                            r.tz_offset = '+4800'; // make sure we have a mismatch
+                        },
+                    },
+                },
+            },
+        };
+
+        await makeView({
+            type: 'list',
+            resModel: 'partner',
+            serverData,
+            resId: 1,
+            arch: /*xml*/ `
+                <tree string="Colors" editable="top">
+                    <field name="tz_offset" invisible="True"/>
+                    <field name="color" widget="timezone_mismatch" />
+                </tree>
+            `,
+        });
+
+        assert.strictEqual($(target).find('td:contains(Red)').length, 3, 'should have 3 rows with correct value');
+        await click($(target).find('td:contains(Red)')[0]);
+
+        const $td = $(target).find('tbody tr.o_selected_row td:not(.o_list_record_selector)');
+
+        assert.strictEqual($td.find('select').length, 1, "td should have a child 'select'");
+        assert.strictEqual(
```

Comment:
The original counts the "contents()" of the $td element but instead of getting 1 as length, I'm getting 2. The second item is a #text node.